### PR TITLE
Controller interactions cleanup

### PR DESCRIPTION
**Pull request by exofunctor**

### CODE REVIEW

**Review by PeruDayani — COMMENTED** [review]
Looks awesome, just to be clear. This pull request is only for this one script to map the state of the buttons right? 

The breakdown of Controller interactions into Map, Control, etc will be a different pull request?

File: Assets/Scripts/Interaction Scripts/ControllerState.cs [hunk 1]
```diff
@@ -0,0 +1,195 @@
+﻿namespace ISAACS {
+
+    using System.Collections;
+    using System.Collections.Generic;
+	using VRTK;
```
Comment (erwang01) [inline Assets/Scripts/Interaction Scripts/ControllerState.cs]: add tab.

File: Assets/Scripts/Interaction Scripts/ControllerState.cs [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    using UnityEngine;
+
+	public class ControllerState : MonoBehaviour {
+
+		// Remember to set these controllers to actual GameObjects through the Unity GUI.
+		public GameObject LeftController;
```
Comment (erwang01) [inline Assets/Scripts/Interaction Scripts/ControllerState.cs:17]: What is the difference between LeftController and controller_left? Same for the right ones. Please add documentation comment.

File: Assets/Scripts/Interaction Scripts/ControllerState.cs [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+		private Transform RightOrigin; // The origin ("position") of the right controller.
+		private Vector3 RightVelocity; // How fast the right controller is moving in space.
+		private Vector3 RightAngularVelocity; // How fast the right controller is moving in space.
+
+
+		// Update is called once per frame.
+		void Update () {
+
+			// These variables store the controller state, which can be queried for input information.
+            var left = LeftController.GetComponent<VRTK_ControllerEvents>();
```
Comment (erwang01) [inline Assets/Scripts/Interaction Scripts/ControllerState.cs:49]: Extra tab

File: Assets/Scripts/Interaction Scripts/ControllerState.cs [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+		public GameObject controller_right;
+
+		// True indicates that the button is being pressed, False that it's released.
+		private bool LeftIsGrabbing; // Whether the left hand is grabbing another object.
+		private bool LeftIndex;
+		private bool LeftMiddle;
+		private bool LeftX;
+		private bool LeftY;
+		private bool LeftThumb;
+		private Vector2 LeftThumbDelta; // The distance by which the left stick has been moved in the X-axis and Y-axis.
```
Comment (erwang01) [inline Assets/Scripts/Interaction Scripts/ControllerState.cs]: Minor nit pick but these description comments can be added to the Getter methods below as well.

Using C# `///` comments will showup in IDE help displays.

File: Assets/Scripts/Interaction Scripts/ControllerState.cs [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+		{
+			return RightVelocity;
+		}
+
+		public Vector3 GetRightAngularVelocity()
+		{
+			return RightAngularVelocity;
+		}
+
+	}
```
Comment (erwang01) [inline Assets/Scripts/Interaction Scripts/ControllerState.cs:196]: check indentation idt this is 4 spaces to a tab.

**Review by erwang01 — CHANGES_REQUESTED** [review]
Generally good from a code writing point of view. Add some comments and fix indentation.

**Comment by exofunctor** [comment]
@PeruDayani yes, this PR is only for the script. I will however push the breakdown and changes in the main unity scene on subsequent PRs (in this same branch).

**Comment by exofunctor** [comment]
@erwang01 thanks for the review, I'll integrate your changes.

File: Assets/Scripts/Interaction Scripts/ControllerState.cs (hunk 5 above)
Comment (exofunctor) [reply Assets/Scripts/Interaction Scripts/ControllerState.cs:196]: I think GitHub hides the namespace brackets. There's normally one more bracket below.

**Review by erwang01 — APPROVED** [review]
Thanks

**Comment by PeruDayani** [comment]
> @PeruDayani yes, this PR is only for the script. I will however push the breakdown and changes in the main unity scene on subsequent PRs (in this same branch).

Awesome, good to merge for me too then!

**Comment by erwang01** [comment]
@apollovision want to do the honors?

**Comment by PeruDayani** [comment]
Keeping the branch open for future pull requests for the breakdown of Controller interactions into Map, Control, etc.

**Pull request merged**